### ISSUE TITLE
org.zalando.logbook.httpclient.LocalRequest replaces body of HttpEntityEnclosingRequest incorrectly

### ISSUE THREAD

**Issue by intuitiveminds** [zalando/logbook#775]
## Description
I am trying to send a request using Content-Type "application/x-www-form-urlencoded" using Apache HttpClient 4.5.12. **Without** logbook, the following request is sent:

```
{
  "args": {}, 
  "data": "", 
  "files": {}, 
  "form": {
    "client_id": "id", 
    "grant_type": "password", 
    "password": "secret", 
    "username": "username"
  }, 
  "headers": {
    "Accept-Encoding": "gzip,deflate", 
    "Authorization": "Basic xxx", 
    "Connection": "Keep-Alive", 
    "Content-Length": "82", 
    "Content-Type": "application/x-www-form-urlencoded", 
    "Host": "127.0.0.1:5080", 
    "User-Agent": "Apache-HttpClient/4.5.12 (Java/1.8.0_242)"
  }, 
  "json": null, 
  "origin": "172.17.0.1", 
  "url": "http://127.0.0.1:5080/post"
}
```

**With** logbook, the request sent looks like this:
```
{
  "args": {}, 
  "data": "grant_type=password&client_id=id&username=username&password=secret", 
  "files": {}, 
  "form": {}, 
  "headers": {
    "Accept-Encoding": "gzip,deflate", 
    "Authorization": "Basic xxx, 
    "Connection": "Keep-Alive", 
    "Content-Length": "82", 
    "Host": "127.0.0.1:5080", 
    "User-Agent": "Apache-HttpClient/4.5.12 (Java/1.8.0_242)"
  }, 
  "json": null, 
  "origin": "172.17.0.1", 
  "url": "http://127.0.0.1:5080/post"
}
```

The "culprit" is in LocalRequest:
```
    private static final class Offering implements State {

        @Override
        public State without() {
            return new Unbuffered();
        }

        @Override
        public State buffer(final HttpRequest request) throws IOException {
            if (request instanceof HttpEntityEnclosingRequest) {
                final HttpEntityEnclosingRequest original = (HttpEntityEnclosingRequest) request;
                if (original.getEntity() == null) {
                    return new Passing();
                } else {
                    final byte[] body = toByteArray(original.getEntity());
                    original.setEntity(new ByteArrayEntity(body));
                    return new Buffering(body);
                }
            } else {
                return new Passing();
            }
        }

    }
```

Regardless of the original type of HTTP entity, it is always replaced with a ByteArray.

## Expected Behavior
The provided body and HTTP entity should be transmitted as-is.

## Actual Behavior
The HTTP entity is changed

## Possible Fix
Either allow a verbatim 1:1 transmission or allow to extend this mechanism via a customizable request copy mechanism.

## Steps to Reproduce
```
HttpPost post = new HttpPost("https://httpbin.org/post");

try {
	post.setEntity(new UrlEncodedFormEntity("foo", "bar"));
} catch (UnsupportedEncodingException ex) {
	// ignore
}

CloseableHttpResponse response = httpClient.execute(post)
```

and check the response.

## Context
Used within a Spring Boot application, using the Logbook instance from the Spring Boot starter as-is (via dependency injection).

## Your Environment
* Version used: 2.1.0

**Comment by whiskeysierra** [comment]
I don't think the type of entity is the issue here. We are missing to set the proper content type, it seems.

**Comment by whiskeysierra** [comment]
I just released 2.1.2 with a fix.
